server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -83,8 +85,18 @@ private static NumberFieldMapper toType(FieldMapper in) {
 
         private final Parameter<Script> script = Parameter.scriptParam(m -> toType(m).builder.script.get());
         private final Parameter<String> onScriptError = Parameter.onScriptErrorParam(m -> toType(m).onScriptError, script);
+
+        /**
+         * Parameter that marks this field as a time series dimension.
+         */
         private final Parameter<Boolean> dimension;
 
+        /**
+         * Parameter that marks this field as a time series metric defining its time series metric type.
+         * For the numeric fields gauge and counter metric types are
+         * supported
+         */
+        private final Parameter<String> metric;
```

Comment:
Should this be `Parameter<MetricType>`? `Setting`s can do things like this. I'm not sure about `Parameter`.